frontend/src/modules/textfilter/textfilter.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import { FormField, Grid, Input, TextContent, TextFilterProps } from '@cloudscape-design/components';
+import { BaseKeyDetail, CancelableEventHandler } from '@cloudscape-design/components/internal/events';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import Condition from '../condition';
+import { debounce } from 'lodash';
+
+export type MLSTextFilterProps = TextFilterProps & {
+    onKeyDown?: CancelableEventHandler<BaseKeyDetail>,
+    requireEnter?: boolean,
+    delay?: number
+};
+
+const KEYCODE_ENTER = 13;
+const KEYCODE_ESC = 27;
+
+export function MLSTextFilter (props: MLSTextFilterProps) {
+    const { requireEnter = false, delay = 200, onKeyDown, onChange, onDelayedChange, filteringText } = props;
+
+    const [state, setState] = React.useReducer((state, action: Partial<{dirty: boolean, filteringText: string}>) => {
+        return {
+            ...state,
+            ...action
+        };
+    }, {
+        dirty: false as boolean,
+        filteringText: props.filteringText
+    });
+
+    useEffect(() => {
+        setState({filteringText});
+    }, [filteringText]);
+
+    // memoize the callback with useCallback
+    const memoizeOnDelayedChange = useCallback((event) => {
```

Comment:
This should maybe be a utility. We use debouncing in translate-realtime and may be worth consolidating/standardizing

Reply:
I'll open a backlog item for this.